src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction fruitTransaction: transactions) {
+            operationStrategy.get(fruitTransaction.getOperation()).updateDataBase(fruitTransaction);
+        }
```

Comment:
Consider adding error handling within the loop to manage potential exceptions that might occur during the processing of each transaction. This will help ensure that one faulty transaction does not disrupt the entire processing sequence.